convert/src/label_studio_to_badgerdoc/labelstudio_format/label_studio_format.py

```diff
@@ -105,3 +156,139 @@ def form_link_direction(self, link: AnnotationLink) -> str:
+        )
+
+        try:
+            request_to_get_taxonomy = requests.get(
+                url=get_taxonomy_url,
+                headers=request_headers,
+            )
+            request_to_get_taxonomy.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            LOGGER.exception(
+                "Failed request to 'taxonomy' to get corresponding taxonomy"
+            )
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail="Failed request to 'taxonomy' to get corresponding taxonomy",
+            ) from e
+        response_content = request_to_get_taxonomy.json()
+        LOGGER.debug(
+            "Got this response from taxonomy service: %s", response_content
+        )
+
+        return [
+            {"taxonomy_id": element["id"], "version": element["version"]}
+            for element in response_content
+        ]
+
+    def create_categories_to_taxonomy_mapping(
+        self, job_id, categories_linked_with_taxonomies, request_headers
+    ):
+        result = {}
+        for category_id in categories_linked_with_taxonomies:
+            taxonomy_objs = self.get_corresponding_taxonomy_obj(
+                job_id=job_id,
+                category_id=category_id,
+                request_headers=request_headers,
+            )
+            for taxonomy_obj in taxonomy_objs:
+                result.update({category_id: taxonomy_obj})
+        return result
+
+    def get_taxonomy_to_taxons_mapping(
+        self,
+        all_taxonomies_ids_used: List[str],
+        request_headers: Dict[str, str],
+    ) -> Dict[str, Any]:
+        get_taxons_used_url = f"{settings.taxonomy_service_url}taxons/search"
+        LOGGER.debug(
+            "Making request to url %s to get all taxons used",
+            get_taxons_used_url,
+        )
+        get_taxons_used_body = {
+            "pagination": {"page_num": 1, "page_size": 100},
+            "filters": [
+                {
+                    "field": "taxonomy_id",
+                    "operator": "in",
+                    "value": all_taxonomies_ids_used,
+                }
+            ],
+        }
+
+        try:
+            request_to_get_taxons_used = requests.post(
+                url=get_taxons_used_url,
+                json=get_taxons_used_body,
+                headers=request_headers,
+            )
+            request_to_get_taxons_used.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            LOGGER.exception("Failed request to 'taxonomy' to get taxons_used")
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail="Failed request to 'taxonomy' to get taxons_used",
+            ) from e
+        response_content = request_to_get_taxons_used.json()
+        LOGGER.debug(
+            "Got this response from taxonomy service: %s", response_content
+        )
```

Comment:
Exact request code repeats 3 times. Probably use a common function?

Reply:
In this file we have 1 GET and 2 POST requests and would prefer to keep it this way now.
These functions are large now because we create url variable, request body and relevant custom logging messages.
On the one hand - there are not a lot of variations and these pieces of code seem to be very close, but on the other hand - this is very easy in debugging and understanding where the problem is. I think this functionality will need a lot of debugging in near future